File: App/Java/app/src/main/java/com/example/tonjies/templateproject/module/weather/presenter/WeatherPresenter.java
package com.example.tonjies.templateproject.module.weather.presenter;

import com.example.common.util.L;
import com.example.tonjies.templateproject.module.weather.bean.Weather;
import com.example.tonjies.templateproject.module.weather.model.WeatherModel;
import com.example.tonjies.templateproject.module.weather.presenter.contract.WeatherContract;

import io.reactivex.Observer;
import io.reactivex.android.schedulers.AndroidSchedulers;
import io.reactivex.disposables.Disposable;
import io.reactivex.schedulers.Schedulers;

/**
 * Created by 舍长 on 2019/1/5
 * describe:
 */
public class WeatherPresenter implements WeatherContract.WeatherIPresenter {

    private WeatherContract.WeatherView weatherView;
    private WeatherModel weatherModel;

    public WeatherPresenter(WeatherContract.WeatherView weatherView) {
        this.weatherView = weatherView;
        weatherModel = new WeatherModel();
    }

    //请求网络数据
    @Override
    public void requestWeather(final String weatherId) {
        L.d("requestWeather");
        weatherModel.requestWeather(weatherId)
                .subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe(new Observer<Weather>() {
                    @Override
                    public void onSubscribe(Disposable d) {

                    }

                    @Override
                    public void onNext(Weather weather) {
                        L.d("获取到的天气信息:" + weatherId);
                        weatherView.getData(weather);
                    }

                    @Override
                    public void onError(Throwable e) {
                        L.d("----"+e.toString());
                    }

                    @Override
                    public void onComplete() {

                    }
                });
    }
}

File: App/kotlins/app/src/main/java/com/example/tonjies/weatherx/module/weather/presenter/WeatherPresenter.kt
package com.example.tonjies.weatherx.module.weather.presenter

import com.example.tonjies.weatherx.base.BasePresenter
import com.example.tonjies.weatherx.ext.BaseSubscriber
import com.example.tonjies.weatherx.ext.exexute
import com.example.tonjies.weatherx.module.weather.bean.Weather
import com.example.tonjies.weatherx.module.weather.view.WeatherView
import com.example.tonjies.weatherx.net.NetUtil
import com.example.tonjies.weatherx.util.L

class WeatherPresenter : BasePresenter<WeatherView>() {

    /**
     * 查询天气信息
     */
    fun requestWeather(weatherId: String) {
        var netUtil = NetUtil()
        netUtil.weatherInfo(weatherId)
                .exexute(object : BaseSubscriber<Weather>() {
                    override fun onNext(t: Weather) {
                        val s: String = t.toString()
                        L.d("${s.toString()}")
                        mView.setData(t)
                    }

                    override fun onError(e: Throwable) {

                        super.onError(e)
                    }
                })
//                .exexute(object : BaseSubscriber<ResponseBody>() {
//                    override fun onNext(t: ResponseBody) {
//                        val s: String = t.string()
////                        L.d("$s")
//                        mView.setData(s)
//                    }
//
//                    override fun onError(e: Throwable) {
//                        super.onError(e)
//                        L.d("当前错误$e")
//                    }
//                })
    }
}